src/main/scala/org/ergoplatform/nodeView/wallet/requests/TransactionSigningRequest.scala

```diff
@@ -0,0 +1,71 @@
+package org.ergoplatform.nodeView.wallet.requests
+
+import io.circe.{Decoder, Encoder, Json}
+import org.ergoplatform.http.api.ApiCodecs
+import org.ergoplatform.modifiers.mempool.UnsignedErgoTransaction
+import org.ergoplatform.wallet.secrets.{DhtSecretWrapper, DlogSecretWrapper, PrimitiveSecretKey}
+
+/**
+  * Basic trait for externally provided hint for interpreter (to be used during script execution and signature
+  * generation).
+  */
+trait Hint
+
+/**
+  * Externally provided secret (to be used once for a transaction to sign)
```

Comment:
Consider clarifying what "used once" means. Does it mean the secret cannot be used to sign two transactions. How it is implemented?

Reply:
done